### PR TITLE
Improve handling of property arrays in .ang writer

### LINKED ISSUE #189: .ang writer cannot write property arrays (image quality etc.) over varying shape
Say I have a crystal map with one image quality value per point, but 20 best matching score values per point. Passing `index=0` to the .ang writer is necessary to write the scores to e.g. the pattern fit column in the .ang file, but then the image quality values cannot be written as well. Not passing `index` throws an error when trying to write the best matching score, but *can* write the image quality array. Should not be difficult to fix this.

Also, the column names in the .ang file header doesn't correspond to the actual data column.

I hope to fix this within the week, but might have to be pushed for a v0.6.1 patch.

### PR DESCRIPTION
**Pull request by hakonanes**
#### Description of the change
Fix #189.

#### Progress of the PR
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [x] Unit tests with pytest for all lines
- [x] Clean code style by [running black via pre-commit](https://orix.readthedocs.io/en/latest/contributing.html#code-style)

#### For reviewers

- [x] The PR title is short, concise, and will make sense 1 year later.
- [x] New functions are imported in corresponding `__init__.py`.
- [x] New features, API changes, and deprecations are mentioned in the
      unreleased section in `CHANGELOG.rst`.

### CODE REVIEW

**Review by pc494 — APPROVED** [review]
LGTM

**Comment by hakonanes** [comment]
Thanks! I'll create release notes tagged off of master for people to review after this is merged.

File: orix/io/plugins/ang.py [hunk 1]
```diff
@@ -636,29 +638,29 @@ def _get_prop_arrays(xmap, prop_names, desired_prop_names, map_size, index, deci
...
     # in the loop
     all_expected_prop_names = [
-        ["iq", "image_quality", "imagequality"],
-        ["ci", "confidence_index", "confidenceindex"],
-        ["ss", "sem_signal", "semsignal"],
-        ["fit", "pattern_fit", "patternfit"],
+        ["iq", "imagequality"],
+        ["ci", "confidenceindex", "scores", "correlation"],
+        ["ss", "semsignal", "detectorsignal"],
+        ["fit", "patternfit"],
```
Comment (harripj) [inline orix/io/plugins/ang.py:652]: I haven't used this functionality before, so just an off-comment. `detector_signal` and `confidence_index` here have underscores, but in line 508-9 they do not. Is this expected or should they be the same?

Comment (hakonanes) [reply orix/io/plugins/ang.py:652]: The column names in the .ang file header are just for readability, and aren't used in any software as far as I know. In the shown lines we assume underscores and spaces are stripped away in order to find them among `CrystalMap.prop`.

Comment (harripj) [reply orix/io/plugins/ang.py:652]: Thanks for clarifying!

**Comment by harripj** [comment]
Merging.

**Comment by hakonanes** [comment]
Thanks!

**Pull request merged**